src/devices/Button/ButtonBase.cs

```diff
@@ -78,17 +80,24 @@ public ButtonBase()
         /// </summary>
         /// <param name="doublePress">Max ticks between button presses to count as doublepress.</param>
         /// <param name="holding">Min ms a button is pressed to count as holding.</param>
-        public ButtonBase(TimeSpan doublePress, TimeSpan holding)
+        /// <param name="debounceTime">The amount of time during which the transitions are ignored, or zero</param>
+        public ButtonBase(TimeSpan doublePress, TimeSpan holding, TimeSpan debounceTime = default(TimeSpan))
         {
             _doublePressTicks = doublePress.Ticks;
             _holdingMs = (long)holding.TotalMilliseconds;
+            _debounceTime = debounceTime;
         }
 
         /// <summary>
         /// Handler for pressing the button.
         /// </summary>
         protected void HandleButtonPressed()
         {
+            if (DateTime.UtcNow.Ticks - _singlePressTicks < _debounceTime.Ticks)
```

Comment:
- Store UtcNow in the beginning of the method and reuse it later
- when using UtcNow anywhere please make it as a sole element on one side of equation (i.e. `UtcNow.Ticks < ...`) - that will make it easier to read

Reply:
In the pressed handler, there is only once. This is why I didn't store it. Do you really want it locally?
In the released handler, I need it before raising the event while the previous code (double press) comes after the event and some significative time may have passed. It would be risky to reuse the same.